content/developer/addons/addon-quickstart.md

```diff
@@ -0,0 +1,233 @@
+
+#### Define basic info
+
+The `name` parameter is optional; it will default to the slug if omitted. To include special characters here or in the description, use their [HTML entity code](https://www.w3schools.com/html/html_entities.asp).
+
+Provide a great `description` that briefly explains what your addon does from the users' perspective. For `version`, familiarize yourself with [semantic versioning](http://semver.org/).
+
+The `authors` parameters are at your discretion. We recommend using a support address for both the email and URL. `authors` takes an array so be sure to include anyone who's name you want on the addon.
+
+The `mobileFriendly` parameter allows the addon to be automatically disabled when accessing the site via a mobile device. We recommend defining it as `true`. If omitted, it defaults to `false`.
+
+All further optional parameters described below default to `false` if not defined.
+
+#### Define a settings page
+
+To create a "Settings" button that will appear on the addon after it is enabled, add these to your definition:
+
+```json
+"settingsUrl": "/settings/somepagehere",
+"settingsPermission": "Garden.Settings.Manage",
+```
+
+#### Define requirements
+
+You can require a certain version of Vanilla before your addon can be enabled. Require a certain version of Vanilla by adding this:
+
+```json
+"require": {
+    "vanilla": ">=2.4"
+},
+```
+
+You can require other addons to be enabled be using this instead:
+
+```json
+"require": {
+    "vanilla": ">=2.4",
+    "Akismet": ">=1.0.1",
+    "StopForumSpam": ">=1.0"
+},
+```
+
+These checks only apply to addon's being enabled/disabled in the dashboard. If a user manually enables your addon or disables a dependancy in the configuration these checks may not apply. Therefore, it is very important to **use defensive programming techniques to guard against missing prerequisites**, rather than simply assuming they will always be there just because you put it in the requirements.
+
+#### Visibility
+
+If you've got [Vanilla Cloud](http://vanillaforums.com), make sure to also [set your addon's visibility](/developer/adddons/addon-visibility).
+
+#### Define new permissions
+
+Adding new permissions via addon is easy. Any defined here will be added as soon as the addon is enabled. It's important to know more about [using permissions in Vanilla](/developers/framework/permissions/) before doing this.
+
+You can provide an array of permission names using dot syntax. You can optionally use key/value pairs to set a default (`1` will give all roles the permission; `0` is the default and leaves it off for all roles to start).
+
+```json
+"registerPermissions": {
+    "FancyAddon.DoStuff.Add" => 1
+},
+```
+
+Or set the default to whether or not the role currently has an existing permission:
+
+```json
+"registerPermissions": {
+    "Plugins.FancyAddon.PermissionName": "Garden.Settings.Manage"
```

Comment:
Plugins.* is now deprecated. It should be `AddonName.Target.ImperativeVerb` (most of the time)